js/reference.js

```diff
@@ -3301,52 +3361,66 @@
                 }
             };
 
-            columns = self.generateColumnsList(tuple);
-
-            // columns
-            columns.forEach(function (col, i) {
-                addColumn(col, i, false, false);
+            var addInlineColumn = function (col, i) {
+                if (isInline(col)) {
+                    requests.push({inline: true, index: i});
+                } else {
+                    addColToActiveList(col, false, COLUMN_TYPE, i);
+                }
 
                 col.waitFor.forEach(function (wf) {
-                    addColumn(wf, i, true, false);
+                    addColToActiveList(wf, true, COLUMN_TYPE, i);
+                });
+            };
+
+
+            // THE CODE STARTS HERE:
+
+            columns = self.generateColumnsList(tuple);
+
+            // citation
+            if (isDetailed && self.citation) {
+                self.citation.waitFor.forEach(function (col) {
+                    addColToActiveList(col, true, CITATION_TYPE);
                 });
+            }
+
+            // columns without aggregate
+            columns.forEach(function (col, i) {
+                if (!hasAggregate(col)) {
+                    addInlineColumn(col, i);
+                }
             });
 
-            //fkeys
-            sds.fkeys.forEach(function (fk) {
-                if (fk.name in consideredOutbounds) return;
-                consideredOutbounds[fk.name] = true;
-                allOutBounds.push(new ForeignKeyPseudoColumn(self, fk));
+            // columns with aggregate
+            columns.forEach(function (col,i ) {
+                if (hasAggregate(col)) {
+                    addInlineColumn(col, i);
+                }
```

Comment:
Looking at these 2 loops, I'm not sure what's different except the condition checking the opposite in each one.

Reply:
Yup. that's the only difference. The first one will process all the columns that are not aggregate or don't have aggregate in their `wait_for` definition. The other function will take care of the rest of the columns.